### PR TITLE
Fix unnecessary log.Fatals and os.Exits

### PR DESCRIPTION
**Pull request by thrasher-**
# Description

Cleans up unnecessary log.Fatals around the codebase where neccessary. There still exists some for exchange loading via Setup but these have all been fixed in engine branch.

Also moves the os.Exit(1) if number of loaded exchanges == 0 from SetupExchanges to main, so developers don't have to look through the code too far if they find out their programs exits on them unexpectedly. 

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

go test ./... -race

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool 
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally and on Travis with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: communications/telegram/telegram.go [hunk 1]
```diff
@@ -43,9 +44,16 @@ const (
 	talkRoot = "GoCryptoTrader bot"
 )
 
+var (
+	// ErrWaiter is the default timer to wait if an err occurs
+	// before retrying after successfully connecting
+	ErrWaiter = time.Second * 30
```
Comment (gloriousCode) [inline communications/telegram/telegram.go]: I don't think this needs to be exported

File: currency/storage.go [hunk 2]
```diff
@@ -116,15 +116,20 @@ func (s *Storage) RunUpdater(overrides BotOverrides, settings *MainConfiguration
...
 			Name:        settings.CryptocurrencyProvider.Name,
 			Enabled:     true,
 			AccountPlan: settings.CryptocurrencyProvider.AccountPlan,
 			APIkey:      settings.CryptocurrencyProvider.APIkey,
 			Verbose:     settings.CryptocurrencyProvider.Verbose,
 		})
-
-		s.currencyAnalysis = c
+		if err != nil {
+			log.Errorf("Unable to setup CoinMarketCap analysis. Error: %s", err)
```
Comment (gloriousCode) [inline currency/storage.go]: Golang loves lowercase starting error messages when it isn't a acronym or pronoun

File: main.go [hunk 3]
```diff
@@ -124,6 +124,9 @@ func main() {
 	log.Debugf("Global HTTP request timeout: %v.\n", common.HTTPClient.Timeout)
 
 	SetupExchanges()
+	if len(bot.exchanges) == 0 {
+		log.Fatal("No exchanges were able to be loaded. Exiting")
```
Comment (gloriousCode) [inline main.go]: Same here, but its a fatal, and its more go having the issue, not me :D

File: currency/storage.go (hunk 2 above)
Comment (shazbert) [reply currency/storage.go]: https://github.com/golang/go/wiki/Errors; for ref, it does say that, but seeing as this is used as a prefix for the err, all that is needed is the underlying exported error type should be lower case?

Comment (xtda) [reply currency/storage.go]: To add a little more

The reason behind this comes from https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Normally you would pass the error down the chain until its finally displayed

so you may end up with something 
```go
return fmt.Errorf("Something bad")
```

then later down the track
```go 
log.Errorf("Some error: %v", err)
```

That will cause it to display 
```
Some error: Something bad
```

if you are just logging the output it's fine but Goland code detection does not pick up on this and just warns either way (same with some other linters)

File: communications/telegram/telegram.go (hunk 1 above)
Comment (thrasher-) [reply communications/telegram/telegram.go]: This is intentionally done for customisability is engine and for standalone apps

File: currency/storage.go (hunk 2 above)
Comment (thrasher-) [reply currency/storage.go]: This is just the prefix of displaying an error. All throughout our code base we use capital letters for the initial prefix of things and errors being lower case as per go spec, that's why our linters don't mind.

Comment (gloriousCode) [reply currency/storage.go]: An issue for the engine branch then 🦀 😄

File: currency/coinmarketcap/coinmarketcap_test.go [hunk 4]
```diff
@@ -41,7 +41,14 @@ func TestSetup(t *testing.T) {
 	cfg.Enabled = true
 	cfg.AccountPlan = "basic"
 
-	c.Setup(cfg)
+	if err := c.Setup(cfg); err != nil {
+		t.Error(err)
+	}
+
+	cfg.AccountPlan = "meow"
```
Comment (shazbert) [inline currency/coinmarketcap/coinmarketcap_test.go]: :100:

**Pull request merged**